src/libcore/str/mod.rs

```diff
@@ -382,6 +382,34 @@ pub unsafe fn from_utf8_unchecked(v: &[u8]) -> &str {
 /// See the immutable version, [`from_utf8_unchecked()`][fromutf8], for more information.
 ///
 /// [fromutf8]: fn.from_utf8_unchecked.html
+///
+/// # Examples
+///
+/// Basic usage:
+///
+/// ```
+/// use std::str;
+///
+/// let mut heart = vec![240, 159, 146, 150];
+/// let heart = unsafe { str::from_utf8_unchecked_mut(&mut heart) };
+///
+/// assert_eq!("💖", heart);
+/// ```
+///
+/// Invalid UTF-8:
+///
+/// ```
+/// use std::str;
+///
+/// // Invalid bytes.
+/// let mut bytes = vec![240, 40, 140, 188];
+///
+/// // Returns a str:
+/// unsafe { str::from_utf8_unchecked_mut(&mut bytes) };
```

Comment:
it turns out that constructing an invalid string in an 'unchecked' manner is considered "undefined behavior" and not something we want to add in here. although it might not be obvious for such a simple example as this, this sort of behavior can cause the program to become unpredictable and problematic. unfortunately, i think we might just need to remove this "Invalid UTF-8" section for this example